building-applications/flare-pattern.md

```diff
@@ -0,0 +1,318 @@
+# Active Retrieval Augmented Generation (FLARE) Pattern
+
+The Flare pattern is an extension of the [The Retrieval Augmented Generation (RAG) pattern](./rag-pattern.md) that adds a feedback loop to improve the quality of the answers provided by the LLM.
+
+You can read the original paper describing the FLARE pattern [here](https://arxiv.org/abs/2305.06983).
+
+Please note that currently FLARE can be implemented only with OpenAI models and the [ai-text-completions](../pipeline-agents/ai-actions/ai-text-completions.md) agent because it requires the probability of correctness (logprobs) for each token.
```

Comment:
```suggestion
Please note that FLARE can currently be implemented only with OpenAI models and the [ai-text-completions](../pipeline-agents/ai-actions/ai-text-completions.md) agent because it requires the probability of correctness (logprobs) for each token.
```